docs/design/features/globalization-icu.md

```diff
@@ -0,0 +1,131 @@
+
+Applications can enable NLS mode by either of the following:
+
+1. in project file:
+
+```xml
+<ItemGroup>
+  <RuntimeHostConfigurationOption Include="System.Globalization.AppLocalIcu" Value="<suffix>:<version> or <version>" />
+</ItemGroup>
+```
+
+2. in `runtimeconfig.json` file:
+
+```json
+{
+  "runtimeOptions": {
+     "configProperties": {
+       "System.Globalization.AppLocalIcu": "<suffix>:<version> or <version>"
+      }
+  }
+}
+```
+
+3. setting environment variable value `DOTNET_SYSTEM_GLOBALIZATION_APPLOCALICU` to `<suffix>:<version>` or `<version>`. 
+
+`<suffix>`: this is optional in the config property, but this follows the public ICU packaging conventions as when building a custom ICU you can customize it to produce the lib names and exported symbol names to contain a suffix. i.e: `libicuucmyapp` where `myapp` is the suffix. This can't be greater than 35 chars in length for the config switch.
+`<version>`: this has to be a valid ICU version, i.e: 67.1. This version will be used to load the binaries and to get the exported symbols.
+
+To load ICU when the app-local switch is set, we use `NativeLibrary.TryLoad` api which does probing in different paths, first it tries to find the library in `NATIVE_DLL_SEARCH_DIRECTORIES` property which is created by the dotnet host based on the `deps.json` file for the app. More details [here](https://docs.microsoft.com/en-us/dotnet/core/dependency-loading/default-probing)
+
+For self contained apps, the user doesn't really need to do anything special, other than making sure ICU is side by side in the APP directory, this is because for self-contained apps, it's work directory is by default in `NATIVE_DLL_SEARCH_DIRECTORIES`. 
+
+If you're consuming ICU via a NuGet package, this will work in framework-dependent apps as NuGet will resolve the native assets and include them in the `deps.json` file and in the output directory for the app under the `runtimes` dir, then we will load it from there.
+
+The tricky part comes whenever it is a framework-dependent app (not self contained) and ICU is consumed from a local build. The SDK doesn't yet have a feature for "loose" native binaries to land into `deps.json`: https://github.com/dotnet/sdk/issues/11373. 
+
+However, there is a workaround by adding something like this to the csproj:
+
+```xml
+<ItemGroup>
+  <IcuAssemblies Include="icu\*.so*" />
+  <RuntimeTargetsCopyLocalItems Include="@(IcuAssemblies)" AssetType="native" CopyLocal="true" DestinationSubDirectory="runtimes/linux-x64/native/" DestinationSubPath="%(FileName)%(Extension)" RuntimeIdentifier="linux-x64" NuGetPackageId="System.Private.Runtime.UnicodeData" />
+</ItemGroup>
+```
+
+Note that this will have to be done for all the ICU binaries for the supported runtimes. Also, the `NuGetPackageId` metadata in the `RuntimeTargetsCopyLocalItems` item group, needs to match a NuGet package that the project actually references, it can't just be a dummy NuGet package.
```

Comment:
```suggestion
Note that this will have to be done for all the ICU binaries for the supported runtimes. Also, the `NuGetPackageId` metadata in the `RuntimeTargetsCopyLocalItems` item group needs to match a NuGet package that the project actually references.
```